docs/src/main/asciidoc/elasticsearch.adoc

```diff
@@ -0,0 +1,414 @@
+
+    public void index(Fruit fruit) throws IOException {
+        IndexRequest request = new IndexRequest("fruits"); // <2>
+        request.id(fruit.id);
+        request.source(JsonObject.mapFrom(fruit).toString(), XContentType.JSON); // <3>
+        restHighLevelClient.index(request, RequestOptions.DEFAULT); // <4>
+    }
+
+    public Fruit get(String id) throws IOException {
+        GetRequest getRequest = new GetRequest("fruits", id);
+        GetResponse getResponse = restHighLevelClient.get(getRequest, RequestOptions.DEFAULT);
+        if (getResponse.isExists()) {
+            String sourceAsString = getResponse.getSourceAsString();
+            JsonObject json = new JsonObject(sourceAsString); // <5>
+            return json.mapTo(Fruit.class);
+        }
+        return null;
+    }
+
+    public List<Fruit> searchByColor(String color) throws IOException {
+        return search("color", color);
+    }
+
+    public List<Fruit> searchByName(String name) throws IOException {
+        return search("name", name);
+    }
+
+    private List<Fruit> search(String term, String match) throws IOException {
+        SearchRequest searchRequest = new SearchRequest("fruits");
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(QueryBuilders.matchQuery(term, match));
+        searchRequest.source(searchSourceBuilder);
+
+        SearchResponse searchResponse = restHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
+        SearchHits hits = searchResponse.getHits();
+        List<Fruit> results = new ArrayList<>(hits.getHits().length);
+        for (SearchHit hit : hits.getHits()) {
+            String sourceAsString = hit.getSourceAsString();
+            JsonObject json = new JsonObject(sourceAsString);
+            results.add(json.mapTo(Fruit.class));
+        }
+        return results;
+    }
+}
+----
+
+In this example you can note the following:
+
+1. We inject an Elasticsearch `RestHighLevelClient` inside the service.
+2. We create an Elasticsearch index request.
+3. We use Vert.x `JsonObject` to serialize the object before sending it to Elasticsearch, you can use whatever you want to serialize to JSON.
+4. We send the request to Elasticsearch.
+5. In order to deserialize the object from Elasticsearch, we again use Vert.x `JsonObject`.
+
+== Hibernate Search Elasticsearch
+
+Quarkus supports Hibernate Search with Elasticsearch via the `hibernate-search-elasticsearch` extension.
+
+Hibernate Search Elasticsearch allows to synchronize your JPA entities to an Elasticsearch cluster and offers a way to query your Elasticsearch cluster using the Hibernate Search API.
+
+If you're interested in it, you can read the link:hibernate-search-elasticsearch[Hibernate Search with Elasticsearch guide].
+
+== Cluster Health Check
+
+If you are using the `quarkus-smallrye-health` extension, both the extension will automatically add a readiness health check
```

Comment:
Not sure "readiness" should be mentioned here? Unless Quarkus waits for the health check to pass before bringing the application online, this wording is confusing.

Reply:
That's the appropriate vocabulary. You have readiness and liveness checks.